server/app/com/xsn/explorer/data/anorm/dao/BlockRewardPostgresDAO.scala

```diff
@@ -0,0 +1,84 @@
+package com.xsn.explorer.data.anorm.dao
+
+import java.sql.Connection
+
+import anorm._
+import com.google.inject.Inject
+import com.xsn.explorer.models.{
+  BlockReward,
+  BlockRewards,
+  PoSBlockRewards,
+  PoWBlockRewards,
+  RewardType,
+  TPoSBlockRewards
+}
+import com.xsn.explorer.models.values.Blockhash
+import com.xsn.explorer.data.anorm.parsers.BlockRewardParsers.parseBlockReward
+
+class BlockRewardPostgresDAO @Inject()(blockPostgresDAO: BlockPostgresDAO) {
+
+  def upsert(blockhash: Blockhash, reward: BlockRewards)(implicit conn: Connection): Unit = {
+    reward match {
+      case r: PoWBlockRewards => upsert(blockhash, r.reward, RewardType.PoW)
+      case r: PoSBlockRewards => upsert(blockhash, r.coinstake, RewardType.PoS)
+      case r: TPoSBlockRewards => {
+        upsert(blockhash, r.owner, RewardType.TPoSOwner)
+        upsert(blockhash, r.merchant, RewardType.TPoSMerchant)
+      }
+    }
+  }
+
+  def getBy(blockhash: Blockhash)(implicit conn: Connection): Option[BlockRewards] = {
+    val rewards = getRewards(blockhash)
+    rewards match {
+      case (r: BlockReward, RewardType.PoW) :: Nil =>
+        Some(PoWBlockRewards(r))
+      case (r: BlockReward, RewardType.PoS) :: Nil =>
+        Some(PoSBlockRewards(r, None))
+      case (ownerReward: BlockReward, RewardType.TPoSOwner) :: (merchantReward: BlockReward, RewardType.TPoSMerchant) :: Nil =>
```

Comment:
How can you guarantee that the TPoS owner will always came before the TPoS merchant reward?

Reply:
i could guarantee it with an `order by` in the query, but having two cases for both scenarios seem clearer to me

Reply:
That's the point, it is hard to be sure whether the owner reward will came before the merchant, you can just match against both cases.

Reply:
well. both cases are already there so nothing to do here then?